ts/components/definition.tsx

```diff
@@ -28,7 +28,7 @@ export interface DefinitionProps {
     className?: string;
 }
 
-export const Definition: React.FC<DefinitionProps> = ({ className, ...props }: DefinitionProps) => (
+export const Definition: React.FC<DefinitionProps> = React.memo(({ className, ...props }: DefinitionProps) => (
```

Comment:
These were _super_ laggy due to context threading (I did some perf analysis via the react profiler). They can all be safely memoized, and increases site perf a good bit.